src/v/cloud_storage/segment_meta_cstore.cc

```diff
@@ -1002,7 +1002,98 @@ class segment_meta_cstore::impl
         _write_buffer.erase(it);
     }
 
-    void insert(const segment_meta& m) {
+    bool can_be_inserted(const segment_meta& m) const {
```

Comment:
move the code so that it doesn't get jumbled up with `insert` in the diff. you can probably just move it to the end of the file?

Reply:
done